Source/WebCore/animation/AnimationTimeline.h

```diff
@@ -63,6 +64,7 @@ class AnimationTimeline : public RefCounted<AnimationTimeline>, public CanMakeWe
     virtual void suspendAnimations();
     virtual void resumeAnimations();
     bool animationsAreSuspended() const;
+    virtual Element* originatingElement() const { return nullptr; };
```

Comment:
I'm not a fan of this new method. I understand the idea here but we still use it a lot in conjunction with a type check. At the moment, only scroll and view timelines can be instantiated via CSS in the wrong place, so it seems a bit generic to put it on `AnimationTimeline`. I'd just do away with it entirely and just downcast in the places where we must check this and use `source()` and `subject()` accordingly… or introduce that method on `ScrollTimeline`.

Reply:
Added static function for this.